crates/core/src/blob/packer.rs

```diff
@@ -511,16 +148,8 @@ impl<BE: DecryptWriteBackend> RawPacker<BE> {
     /// # Errors
     ///
     /// * If the packfile could not be saved
-    fn finalize(&mut self) -> RusticResult<PackerStats> {
-        self.save().map_err(|err| {
-            err.overwrite_kind(ErrorKind::Internal)
-                .prepend_guidance_line("Failed to save packfile. Data may be lost.")
-                .ask_report()
-        })?;
-
-        self.file_writer.take().unwrap().finalize()?;
-
-        Ok(std::mem::take(&mut self.stats))
+    pub fn finalize(mut self) -> RusticResult<(Option<(Bytes, IndexPack)>, PackerStats)> {
+        Ok((self.save()?, self.stats))
```

Comment:
Not sure about the error handling here, we would not have an insight here, if we bubble it up just like that, before it was wrapped.

Reply:
Actually, the errors can only occur when writing the pack header in `write_header`. And those are all very well handled errors, IMO. I don't see a reason why we should handle them in another way in `finalize` than we're handling them in `save`...